### PR TITLE
feat: Add installMultipleApks

### PR DESCRIPTION
**Pull request by KazuCocoa**
Closes https://github.com/appium/appium/issues/14823

- multiple apks without options
```ruby
@driver.execute_script 'mobile: installMultipleApks',
                             {
                                 apks: [
                                     'https://github.com/appium/ruby_lib_core/blob/master/test/functional/app/api.apk.zip?raw=true',
                                     'https://github.com/appium/ruby_lib_core/blob/master/test/functional/app/api.apk.zip?raw=true',
                                     'https://github.com/appium/ruby_lib_core/blob/master/test/functional/app/api.apk.zip?raw=true',
                                 ]
                             }
```
```
[debug] [ADB] Running '/Users/kazuaki/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 install-multiple -r /var/folders/y6/524wp8fx0xj5q1rf6fktjrb00000gn/T/2021112-18422-awvpm6.p1l1v/api.apk /var/folders/y6/524wp8fx0xj5q1rf6fktjrb00000gn/T/2021112-18422-awvpm6.p1l1v/api.apk /var/folders/y6/524wp8fx0xj5q1rf6fktjrb00000gn/T/2021112-18422-awvpm6.p1l1v/api.apk'
```
- multiple apks with an option 
```ruby
@driver.execute_script 'mobile: installMultipleApks',
                             {
                                 apks: [
                                     'https://github.com/appium/ruby_lib_core/blob/master/test/functional/app/api.apk.zip?raw=true',
                                     'https://github.com/appium/ruby_lib_core/blob/master/test/functional/app/api.apk.zip?raw=true',
                                     'https://github.com/appium/ruby_lib_core/blob/master/test/functional/app/api.apk.zip?raw=true',
                                 ],
                                 options: {
                                     grantPermissions: true
                                 }
                             }
```
```
[debug] [ADB] Running '/Users/kazuaki/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 install-multiple -r -g /var/folders/y6/524wp8fx0xj5q1rf6fktjrb00000gn/T/2021112-18422-awvpm6.p1l1v/api.apk /var/folders/y6/524wp8fx0xj5q1rf6fktjrb00000gn/T/2021112-18422-awvpm6.p1l1v/api.apk /var/folders/y6/524wp8fx0xj5q1rf6fktjrb00000gn/T/2021112-18422-awvpm6.p1l1v/api.apk'
```

- No apks. Then, it will finish immediately.

```ruby
@driver.execute_script 'mobile: installMultipleApks',
                             {
                                 apks: [],
                                 options: {
                                     grantPermissions: true
                                 }
                             }
```
```
[HTTP] --> POST /wd/hub/session/5681e0f8-1df5-44f8-9fe2-d2fe73ec9411/execute/sync
[HTTP] {"script":"mobile: installMultipleApks","args":[{"options":{"grantPermissions":true}}]}
[debug] [W3C (5681e0f8)] Calling AppiumDriver.execute() with args: ["mobile: installMultipleApks",[{"options":{"grantPermissions":true}}],"5681e0f8-1df5-44f8-9fe2-d2fe73ec9411"]
[AndroidDriver] Executing native command 'mobile: installMultipleApks'
[debug] [UiAutomator2] No apks are given to install
[debug] [W3C (5681e0f8)] Responding to client with driver.execute() result: null
[HTTP] <-- POST /wd/hub/session/5681e0f8-1df5-44f8-9fe2-d2fe73ec9411/execute/sync 200 10 ms - 14
```

### CODE REVIEW

**Comment by KazuCocoa** [comment]
Will update README tomorrow

File: lib/apk-extensions.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+const APK_EXTENSION = '.apk';
```
Comment (mykola-mokhnach) [inline lib/apk-extensions.js:1]: I'd rather rename the file to simply `extensions`

File: lib/commands/general.js [hunk 2]
```diff
@@ -296,6 +300,47 @@ commands.mobileType = async function mobileType (opts = {}) {
...
+
+/**
+ * Install multiple APKs with `install-multiple` option.
+ *
+ * @param {InstallMultipleApksOptions} opts
+ * @throws {Error} if an error occured while installing the given APKs.
+ */
+commands.mobileInstallMultipleApks = async function (opts = {}) {
+  if (!_.isArray(opts.apks) && _.isEmpty(opts.apks)) {
+    log.debug('No apks are given to install');
```
Comment (mykola-mokhnach) [inline lib/commands/general.js:336]: I would rather throw an illegal argument error here

File: lib/commands/general.js [hunk 3]
```diff
@@ -296,6 +300,46 @@ commands.mobileType = async function mobileType (opts = {}) {
...
+ */
+
+/**
+ * Install multiple APKs with `install-multiple` option.
+ *
+ * @param {InstallMultipleApksOptions} opts
+ * @throws {Error} if an error occured while installing the given APKs.
+ */
+commands.mobileInstallMultipleApks = async function (opts = {}) {
+  if (!_.isArray(opts.apks) && _.isEmpty(opts.apks)) {
```
Comment (mykola-mokhnach) [inline lib/commands/general.js:335]: the condition here should probably be `||`

File: lib/commands/general.js [hunk 4]
```diff
@@ -296,6 +300,46 @@ commands.mobileType = async function mobileType (opts = {}) {
...
+ *                                      if it is already present on the device.
+ * @property {boolean} partialInstall [false] - Install apks partially. It is used for 'install-multiple'.
+ *                                             https://android.stackexchange.com/questions/111064/what-is-a-partial-application-install-via-adb
+ */
+
+/**
+ * @typedef {Object} InstallMultipleApksOptions
+ * @property {Array<string>} apks - The list of APKs to install. Each APK should be a path to a apk
+ *                                  or downloadable URL as HTTP/HTTPS.
+ * @property {InstallOptions} options
```
Comment (mykola-mokhnach) [inline lib/commands/general.js:325]: I assume it does not work this way for docs and it is necessary to copypaste properties

File: README.md [hunk 5]
```diff
@@ -639,6 +639,17 @@ Note that `description` in `page` can be an empty string most likely when it com
...
+
+Install applications via `install-multiple` option.
+Please read more details in the secrion by `adb --help`.
+
+#### Arguments
+
+Name | Type | Required | Description | Example
+--- | --- | --- | --- | ---
+apks | array<string> | yes | The path to APKs. Each path should be the full path to the apk to be installed, or an URL to a remote location. | `['/path/to/local.apk', 'https://github.com/appium/ruby_lib_core/blob/master/test/functional/app/api.apk.zip?raw=true']`
+options | object | Whether enable options in `install-multiple` installation as `boolean`. `allowTestPackages` is `-t`, `useSdcard` is `-s`, `grantPermissions` is `-g`, `replace` is `-r`, `partialInstall` is `-p`. | `{grantPermissions: true, partialInstall: true}`
```
Comment (mykola-mokhnach) [inline README.md:652]: Missing Required column

Comment (mykola-mokhnach) [inline README.md:652]: > Whether enable options in `install-multiple` installation as `boolean`

I cannot understand the meaning of this sentence

File: README.md [hunk 6]
```diff
@@ -639,6 +639,17 @@ Note that `description` in `page` can be an empty string most likely when it com
  }
 ```
 
+### mobile: installMultipleApks
+
+Install applications via `install-multiple` option.
+Please read more details in the secrion by `adb --help`.
```
Comment (mykola-mokhnach) [inline README.md:645]: in the corresponding section of `adb --help` output

File: lib/commands/general.js (hunk 3 above)
Comment (KazuCocoa) [reply lib/commands/general.js]: yea.. i've added a test to ensure it

File: README.md [hunk 7]
```diff
@@ -639,6 +639,17 @@ Note that `description` in `page` can be an empty string most likely when it com
...
+
+Install applications via `install-multiple` option.
+Please read more details in the corresponding section of the `adb --help` command output.
+
+#### Arguments
+
+Name | Type | Required | Description | Example
+--- | --- | --- | --- | ---
+apks | array<string> | yes | The path to APKs. Each path should be the full path to the apk to be installed, or an URL to a remote location. | `['/path/to/local.apk', 'https://github.com/appium/ruby_lib_core/blob/master/test/functional/app/api.apk.zip?raw=true']`
+options | object | no | Options to be enabled in `install-multiple` installation. If you want enable `-g` option, you can specify `grantPermissions` as `{grantPermissions: true}` as `options`. `allowTestPackages` is `-t`, `useSdcard` is `-s`, `replace` is `-r` (`-r` is enabled by default), `partialInstall` is `-p`. | `{grantPermissions: true, partialInstall: true}`
```
Comment (mykola-mokhnach) [inline README.md:652]: I'd simply say `Installation options`.

Comment (mykola-mokhnach) [reply README.md:652]: you can specify -> you could specify

Comment (mykola-mokhnach) [reply README.md:652]: `allowTestPackages` corresponds to `-t`....

**Review by mykola-mokhnach — APPROVED** [review]

**Pull request merged**